crates/reward/src/lib.rs

```diff
@@ -380,16 +382,19 @@ where
         Pallet::<T, I>::deposit_stake(reward_id, amount.to_fixed().ok_or(Error::<T, I>::TryIntoIntError)?)
     }
 
-    fn withdraw_stake(reward_id: &T::RewardId, amount: Balance) -> DispatchResult {
-        Pallet::<T, I>::withdraw_stake(reward_id, amount.to_fixed().ok_or(Error::<T, I>::TryIntoIntError)?)
+    fn withdraw_stake(reward_id: &T::RewardId, maybe_amount: Option<Balance>) -> DispatchResult {
+        let amount = maybe_amount.map_or(Ok(Pallet::<T, I>::stake(reward_id)), |amount| {
+            amount.to_fixed().ok_or(Error::<T, I>::TryIntoIntError)
+        })?;
+        Pallet::<T, I>::withdraw_stake(reward_id, amount)
     }
 }
 
 impl<AccountId, Balance> ModifyStake<AccountId, Balance> for () {
     fn deposit_stake(_: &AccountId, _: Balance) -> DispatchResult {
         Ok(())
     }
-    fn withdraw_stake(_: &AccountId, _: Balance) -> DispatchResult {
+    fn withdraw_stake(_: &AccountId, _: Option<Balance>) -> DispatchResult {
```

Comment:
Not used - let's remove this argument. If we need it in the future we can add it back in then